ext/gmp/gmp.c

```diff
@@ -1350,22 +1350,20 @@ ZEND_FUNCTION(gmp_pow)
 		RETURN_THROWS();
 	}
 
-    double powmax = log((double)ZEND_LONG_MAX);
+	double max_bits = 16000; // TODO: value is very small, but passed current test suite
```

Comment:
`max_bits` is a misnomer now; it's actually something like `max_double_bits`. Maybe someone has an idea for a good name. Alternatively, we could double the value here, and check against `> max_bits / 2.0` below. Shouldn't be a performance issue, if we declare `const` or use a macro.